### ISSUE TITLE
value-no-longhand

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#979]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Rule see which values of properties can be shortest (shorthand).

https://developer.mozilla.org/ru/docs/Web/CSS/Shorthand_properties
Simular rule https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#shorthand

> Which rule, if any, is this issue related to?

e.g. `value-no-longhand`

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  margin: 1px 1px 1px 1px;
}
```

to

```css
.foo {
  margin: 1px;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "value-no-longhand": 1
  }
}
```
Value must be 1, 2, 3 as in scss-lint configuration.

> Which version of stylelint are you using?

e.g. `5.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

no matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no matter

> What did you expect to happen?

Throw error

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing

**Comment by jeddy3** [comment]
This rule seems sensible. 

Can you explain the `1`, `2` or `3` options though? I wasn't able to quite grok their purpose from the scsslint docs.

**Comment by davidtheclark** [comment]
Looks like those are enforcing something different altogether: the length of acceptable shorthands. `margin: 10px` = 1, `margin: 10p 20px` = 2, `margin: 10px 20px 15px` = 3. I think that should probably fall under a separate rule rather than being tangled up with `value-no-longhand`.

**Comment by evilebottnawi** [comment]
@jeddy3 
```
margin: 10px = 1, margin: 10p 20px = 2, margin: 10px 20px 15px = 3
```
```
Can you explain the 1, 2 or 3 options though? I wasn't able to quite grok their purpose from the scsslint docs.
```
@davidtheclark what name for rule is best? I can do PR

**Comment by jeddy3** [comment]
>  I think that should probably fall under a separate rule rather than being tangled up with value-no-longhand.

Agreed. I'm struggling to understand what the use case is for disallowing specific shorthand lengths. Either way, I agree that we make this rule a simple `-no-` boolean one. 

> what name for rule is best?

- Name: `value-no-longhand`.
- Primary option: `true`.
- Secondary options: none
- Message: "Unexpected longhand value"

Sound about right?

**Comment by evilebottnawi** [comment]
@jeddy3 sounds good, how about options about shortest: `1`, `2`, `3`

**Comment by jeddy3** [comment]
I think lets go with just a `true` primary option for now. We can discuss at another time the merits of restricting specific shorthand lengths as *either* secondary options of this rule or a different rule entirely.

**Comment by jeddy3** [comment]
Sorry, I've just realised this already exists as a plugin: https://github.com/timothyneiljohnson/stylelint-value-shorthand

@timothyneiljohnson Would you be interested in porting this rule into core (and renaming it `value-no-longhand`)?

**Comment by evilebottnawi** [comment]
@jeddy3 look good, but not working with `border-radius: 20px / 10px 20px / 10px` and not working with variables(sass, less). But we use as basic this rule

**Comment by jeddy3** [comment]
@evilebottnawi Thanks for looking into it. it'd be great if you and Timothy could work together to create a more feature-rich core rule called `value-no-longhand` that is based on Timothy's plugin.

**Comment by evilebottnawi** [comment]
@jeddy3 i think i can find the time at the end of this week

**Comment by evilebottnawi** [comment]
Rename to `shorthand-property-no-redundant-values`. Merged. Close.
